ocrd_models/ocrd_models/ocrd_page_generateds.py

```diff
@@ -5347,7 +5384,28 @@ def buildChildren(self, child_, node, nodeName_, fromsubclass_=False, gds_collec
             obj_.original_tagname_ = 'UnorderedGroupIndexed'
     def __hash__(self):
         return hash(self.id)
-
+    def get_AllIndexed(self):
+        return sorted(self.get_RegionRefIndexed() + self.get_OrderedGroupIndexed() + self.get_UnorderedGroupIndexed(), key=lambda x : x.index)
+    
+    def clear_AllIndexed(self):
+        ret = self.get_AllIndexed()
+        self.set_RegionRefIndexed([])
+        self.set_OrderedGroupIndexed([])
+        self.set_UnorderedGroupIndexed([])
+        return ret
+    
+    def add_AllIndexed(self, elements):
+        if not isinstance(elements, list):
+            elements = [elements]
+        for element in sorted(elements, key=lambda x : x.index):
```

Comment:
Wouldn't it be prudent to do some `@index` consistency checking vodoo here (like no `elements` possess an already existing or mutually conflicting index)? If we have to raise an exception anyway, then the earlier (nearer to the cause) the better, wouldn't you agree?

Reply:
I think that both `@index` consistency as well as regions only partially listed in a RO are errors and should be raised in the PAGE validator. However, depending on how wide-spread these inconsistencies are, we should mitigate them in a way that no information is lost. That is not an issue here but it is for the region-not-in-RO problem above.

Reply:
> I think that both `@index` consistency as well as regions only partially listed in a RO are errors and should be raised in the PAGE validator.

My point was that the PAGE validator may be too late to be useful. For the case of regions missing in RO, I don't think this even _is_ an error (just a question of how the iterator should behave). And for inconsistent `@index` I believe this is so severe that we should not wait until serialization or even validation, but raise an exception right away (so offending processors get fixed soon), or even provide an API that does not allow making this mistake in the first place. 

For example, instead of `add_AllIndexed` (passing `@indexed` through after sorting) one could name the function `extend_AllIndexed` and rewrite all new elements such that their infimum is increased beyond the old elements' supremum (and then sorting).

Reply:
@kba now you did rename from `add` to `extend` already, but the implementation is yet unchanged. Are you still going to? (If so, exceptions or index mitigation?)

Reply:
index mitigation, implementation will follow in this PR, it's a bit tricky.

Reply:
`extend_AllIndexed(elements, validate_continuity=False` now increments the index of all elements passed. If `validate_continuity` is `True`, validates that lowest element `@index` is greater than the highest existing index (though no check for contiguousness), sorts elements by index and adds them (i.e. the previous behaviour plus validation). Otherwise (the default), elemetns are not sorted and added with incrementing `@index`.